### PR TITLE
connectivity: Extend pod-to-pod encryption tests (for WG)

### PR DESCRIPTION
**Pull request by brb**
CI run https://github.com/cilium/cilium/pull/32995. Only for WG (see https://github.com/cilium/cilium/issues/33168 for IPsec issues).

### CODE REVIEW

File: connectivity/builder/pod_to_pod_encryption.go [hunk 1]
```diff
@@ -19,4 +24,13 @@ func (t podToPodEncryption) build(ct *check.ConnectivityTest, _ map[string]strin
 		WithScenarios(
 			tests.PodToPodEncryption(features.RequireEnabled(features.EncryptionPod)),
 		)
+
+	newTest("pod-to-pod-with-l7-policy-encryption", ct).
+		WithFeatureRequirements(features.RequireEnabled(features.L7Proxy), features.RequireEnabled(features.EncryptionPod)).
```
Comment (giorio94) [inline connectivity/builder/pod_to_pod_encryption.go:29]: I guess also we need the `WithCondition` check as above.

**Review by giorio94 — APPROVED** [review]
Looks good to me, thanks! Just one comment inline.

**Review by jschwinger233 — APPROVED** [review]

**Comment by julianwiedmann** [comment]
@brb want to give this another try, now that https://github.com/cilium/cilium/pull/32683 is in? :)

**Comment by brb** [comment]
Oh! Yep, let's try :crossed_fingers:

**Comment by brb** [comment]
Unfortunately, the new test failed with IPsec enabled :-( https://github.com/cilium/cilium/pull/32995

**Comment by brb** [comment]
MBOI @jschwinger233 ^^

**Comment by brb** [comment]
@julianwiedmann Unfortunately, the test case is for WG only at least for now.

**Review by jschwinger233 — APPROVED** [review]

**Pull request merged**